Initialize ir->branch_table->PC with safe value

ChinYikMing opened the pull request:
If the ra(return address) is 0x0, the
LOOKUP_OR_UPDATE_BRANCH_HISTORY_TABLE will bahave abnormally since calloc initialize ir->branch_table->PC[i] to 0x0. The 0x0 address might be not yet translated to a valid block, thus ir->branch_table->target[i] might be NULL, calling a NULL function pointer cause segmentation fault. It can be solved by initializing ir->branch_table->PC will other value than 0x0. Here, I choose unsigned integer of -1.

Close #461

jserv reviewed (commented):
<!-- github-benchmark-action-comment(start): Benchmarks Summary -->
# Benchmarks

<details>

| Benchmark suite | Current: b65d24c0adb4f2e222a8a70639ce5af36819c15f | Previous: 1e4ef3c8802143795e7f4f4e2fee04f995412793 | Ratio |
|-|-|-|-|
| `Dhrystone` | `5` Average DMIPS over 10 runs | `5` Average DMIPS over 10 runs | `1` |
| `Coremark` | `0.004` Average iterations/sec over 10 runs | `0.004` Average iterations/sec over 10 runs | `1` |

</details>

This comment was automatically generated by [workflow](https://github.com/sysprog21/rv32emu/actions?query=workflow%3ABenchmark) using [github-action-benchmark](https://github.com/marketplace/actions/continuous-benchmark).
<!-- github-benchmark-action-comment(end): Benchmarks Summary -->

ChinYikMing committed:
Initialize ir->branch_table->PC with safe value

If the ra(return address) is 0x0, the
LOOKUP_OR_UPDATE_BRANCH_HISTORY_TABLE will bahave abnormally since
calloc initializes ir->branch_table->PC[i] to 0x0. The address 0x0 might
be not yet translated to a valid block, thus ir->branch_table->target[i]
might be NULL, accessing a NULL pointer causes segmentation fault. It
can be solved by initializing ir->branch_table->PC with other value than
0x0. Here, I choose unsigned integer of -1.

Close #461
